src/compile/axis/encode.ts

```diff
@@ -22,6 +22,27 @@ export function labels(model: UnitModel, channel: PositionScaleChannel, specifie
       }),
       ...specifiedLabelsSpec
     };
+  } else if (
```

Comment:
Should we group common predicates? 

```ts
if (format === undefined &&
    formatType === undefined &&
    channelDefType(fieldOrDatumDef) === 'quantitative') {
   if (isPositionFieldOrDatumDef(fieldOrDatumDef) &&
    fieldOrDatumDef.stack === 'normalize') {
       // For normalized stacked field, always apply 
       // <... your new code block>

   } else if (config.customFormatTypes && config.numberFormatType) {
       // the old block below
   }
}
```

Reply:
Btw, I just realize that it should be more like 

```
if (format === undefined &&
    formatType === undefined &&
    channelDefType(fieldOrDatumDef) === 'quantitative' && 
    config.customFormatTypes
) {
   if (isPositionFieldOrDatumDef(fieldOrDatumDef) &&
    fieldOrDatumDef.stack === 'normalize' && config.normalizedNumberFormatType) {
       // only apply the format here if there is `config.normalizedNumberFormatType` and `config.normalizedNumberFormat` -- otherwise, we can output `labelFormat`

   } else if (config.numberFormatType) {
       // the old block below
   }
}
```